contrib_bots/lib/giphy/giphy.py

```diff
@@ -0,0 +1,128 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# ~/.giphy_config
+
+from __future__ import absolute_import
+from __future__ import print_function
+from six.moves.configparser import SafeConfigParser
+import requests
+import logging
+import sys
+import os
+
+def parse_giphy_config(config_file):
+    config = SafeConfigParser()
+    config.readfp(f, config_file)
```

Comment:
So you are expecting an `f` variable here to exists meaning the `parse_giphy_config` is only usable if there is an `f` variable defined in its context. How about you passed the `f` explicitly instead? Wouldn`t that make this function more testable and pure?